drivers/media/i2c/ov5693.c

```diff
@@ -1577,6 +1577,8 @@ static int ov5693_remove(struct i2c_client *client)
 
 	media_entity_cleanup(&ov5693->sd.entity);
 	v4l2_ctrl_handler_free(&ov5693->ctrl_handler);
+	if (ov5693->otp_size)
+		kfree(ov5693->otp_data);
```

Comment:
Good catch!

Reply:
Is `ov5693->otp_data` `NULL` if `ov5693->otp_size` is zero? In that case you can skip the check and just call `kfree` as `kfree(NULL)` is well defined.

Reply:
@qzed No; the pointer's kzalloced here:

```c
static u8 *ov5693_otp_read(struct i2c_client *client)
{
	u8 *buf;
	int ret;

	buf = devm_kzalloc(&client->dev, (OV5693_OTP_DATA_SIZE + 16), GFP_KERNEL);
	if (!buf)
		return ERR_PTR(-ENOMEM);

<snip>

	return buf;
}
...

So it's 0.

Reply:
Ah, I think this should be `devm_kfree` then.

Reply:
@qzed oops, you're right of course. Actually since I'm just freeing in remove that probably was fine to begin with